src/slice/quicksort.rs

```diff
@@ -510,10 +521,7 @@ where
 
         // SAFETY: `pivot` is a reference to the first element of `v`, so `ptr::read` is safe.
         let tmp = mem::ManuallyDrop::new(unsafe { ptr::read(pivot) });
-        let _pivot_guard = CopyOnDrop {
-            src: &*tmp,
-            dest: pivot,
-        };
+        CopyOnDrop::new(&*tmp, pivot);
```

Comment:
```suggestion
        let _pivot_guard = CopyOnDrop::new(&*tmp, pivot);
```
... and maybe `#[must_use]` the type. :)